### ISSUE TITLE
Proposal to add support for product highlights

### ISSUE THREAD

**Issue by alex-jansen** [schemaorg/schemaorg#3148]
This is a proposal from Google based on our experience consuming schema.org Product markup and working with similar data from online merchants. If it were accepted, it would allow merchants to add product highlights to their product structured data, and allow us and others to better understand products sold online.

## Description

Products advertised or sold online are often displayed not only with a title and description, but also with more detailed information, such as a list of product highlights, which are easy to consume and quick to scan ordered bullet points emphasizing the relevant attributes of a product. For example:

*   _Capture quality 24mp photos and UHD video in an easy to use point and shoot camera with a 2.8" preview screen_
*   _Excellent for underwater photography, submersible up to 16 feet_
*   _Protective plastic cover to protect from scratches_
*   _A great first DSLR camera with enough features for novice photographers of any age_
*   _Uses 2 AAA Batteries (included) and saves videos to a mini SD card (not included)_

To support ordered highlight lists we propose to introduce a new property **/highlights** of type [/ItemList](https://schema.org/ItemList) (for use on [/Product](https://schema.org/Product)). The example given above could then be represented as follows:

```
{
  "@context": "https://schema.org/",
  "@type": "Product",
  "sku": "camera-12345",
  "image": "https://www.example.com/camera.jpg",
  "name": "Nice camera",
  "description": "Compact DSLR camera with pancake lens",
  "gtin": "12345678901234",
  "brand": {
    "@type": "Brand",
    "name": "CameraBrand"
  },
  "color": "black",
  "offers": {
    "@type": "Offer",
    "url": "http://www.example.com/camera_offer",
    "itemCondition": "https://schema.org/NewCondition",
    "availability": "https://schema.org/Backorder",
    "price": "539.99",
    "priceCurrency": "EUR"
  },
  "highlights": {
    "@type": "ItemList",
    "itemListElement": [
      {
        "@type": "ListItem",
        "position": 1,
        "item": "Capture quality 24mp photos and UHD video in an easy to use point and shoot camera with a 2.8\" preview screen"
      },
      {
        "@type": "ListItem",
        "position": 2,
        "item": "Excellent for underwater photography, submersible up to 16 feet"
      },
      {
        "@type": "ListItem",
        "position": 3,
        "item": "Protective plastic cover to protect from scratches"
      },
      {
        "@type": "ListItem",
        "position": 4,
        "item": "A great first DSLR camera with enough features for novice photographers of any age"
      },
      {
        "@type": "ListItem",
        "position": 5,
        "item": "Uses 2 AAA Batteries (included) and saves videos to a mini SD card (not included)"
      }
    ]
  }
}
```

It is important to note that the present proposal suffers from the same disadvantage as for example [#2832](https://github.com/schemaorg/schemaorg/issues/2832), i.e., it requires verbose markup to support explicit ordering of list elements. Eventually (but outside scope of this proposal) it would be beneficial to support ordered elements with syntactic sugar or even an extension of RDFS since ordered elements are a very common need and would benefit from a more concise notation.

**Comment by danbri** [comment]
Thanks! 

Let's tweak https://schema.org/positiveNotes instead, so it's expected on Product.

```json 
{
  "@context": "https://schema.org/",
  "@type": "Product",
  "sku": "camera-12345",
  "image": "https://www.example.com/camera.jpg",
  "name": "Nice camera",
  "description": "Compact DSLR camera with pancake lens",
  "gtin": "12345678901234",
  "brand": {
    "@type": "Brand",
    "name": "CameraBrand"
  },
  "color": "black",
  "offers": {
    "@type": "Offer",
    "url": "http://www.example.com/camera_offer",
    "itemCondition": "https://schema.org/NewCondition",
    "availability": "https://schema.org/Backorder",
    "price": "539.99",
    "priceCurrency": "EUR"
  },
  "positiveNotes": {
    "@type": "ItemList",
    "itemListElement": [
      {
        "@type": "ListItem",
        "position": 1,
        "item": "Capture quality 24mp photos and UHD video in an easy to use point and shoot camera with a 2.8\" preview screen"
      },
      {
        "@type": "ListItem",
        "position": 2,
        "item": "Excellent for underwater photography, submersible up to 16 feet"
      },
      {
        "@type": "ListItem",
        "position": 3,
        "item": "Protective plastic cover to protect from scratches"
      },
      {
        "@type": "ListItem",
        "position": 4,
        "item": "A great first DSLR camera with enough features for novice photographers of any age"
      },
      {
        "@type": "ListItem",
        "position": 5,
        "item": "Uses 2 AAA Batteries (included) and saves videos to a mini SD card (not included)"
      }
    ]
  }
}
```

The definition would also need a few words about Products vs Reviews, but this feels close enough for us to avoid bloating the world with a new property. Does this work for you?

**Comment by danbri** [comment]
> Indicates, in the context of a [Review](https://schema.org/Review) (e.g. framed as 'pro' vs 'con' considerations), positive considerations - either as unstructured text, or a list.

would become

> Provides positive considerations regarding something, for example product highlights or (alongside [[negativeNotes]] for pro/con lists for reviews.
>
> In the case of a Review, the property describes the [[itemReviewed]] from the perspective of the review; in the case of a [[Product]], the product itself is being described.
>
> The property values can be expressed either as unstructured text (repeated as necessary), or if ordered, as a list (in which case the most positive is at the beginning of the list).

**Comment by alex-jansen** [comment]
Thanks for the quick review Dan! With your proposed rephrasing it makes sense to leverage the existing [/positiveNotes](https://schema.org.positiveNotes) property. Eventually we might then even consider expanding its use beyond [/Product](https://schema.org/Product) and [/Review](https://schema.org.Review)

**Comment by danbri** [comment]
Thanks. The only oddity I'll mention but pass over for now, is the subtle shift in terms of authorship. In the Product case it's in the voice of whoever's structured data is expressing things, ... in the Review case, we're summarizing a Review that may be in an entirely different voice. But those topics are a huge can of worms, I think the redrafted description covers a cluster of related usecases quite reasonably.

**Comment by danbri** [comment]
I've just addressed the negativeNotes case, but tagged the commit only against #2832 rather than here and #3149.

**Comment by danbri** [comment]
https://github.com/schemaorg/schemaorg/commit/d5a550f2fe8545d70b2ee56674d4b236608680ec

**Comment by danbri** [comment]
This is ready for staging now.

**Comment by alex-jansen** [comment]
Released in 15.0.
